Core/Src/state_machine.c

```diff
@@ -171,44 +150,37 @@ static int transition_nero_state(nero_state_t new_state)
 	if (!current_nero_state.home_mode && !new_state.home_mode) {
 		// Only Check if we are in pit mode to toggle direction
 		if (current_nero_state.nero_index == PIT) {
-			if (get_drive_state() == REVERSE) {
-				if (transition_drive_state(SPEED_LIMITED))
+			if (get_func_state() == REVERSE) {
+				if (transition_functional_state(F_PIT, pdu, mc))
 					return 1;
-			} else {
-				if (transition_drive_state(REVERSE))
+			} else if (get_func_state() == F_PIT) {
```

Comment:
if this is not true then something has gone horrifiically wrong

Reply:
Same as reply to previous comment about pit and reverse mode regarding my understanding of the relationship between the states.